service.kn.switchtimer/handler.py

```diff
@@ -0,0 +1,161 @@
+import sys
+import time
+import datetime
+import xbmc, xbmcaddon, xbmcgui
+import os
+import operator
+
+__addon__ = xbmcaddon.Addon()
+__addonid__ = __addon__.getAddonInfo('id')
+__addonname__ = __addon__.getAddonInfo('name')
+__path__ = __addon__.getAddonInfo('path')
+__version__ = __addon__.getAddonInfo('version')
+__LS__ = __addon__.getLocalizedString
+
+__IconDefault__ = xbmc.translatePath(os.path.join( __path__,'resources', 'media', 'default.png'))
+__IconAlert__ = xbmc.translatePath(os.path.join( __path__,'resources', 'media', 'alert.png'))
+__IconOk__ = xbmc.translatePath(os.path.join( __path__,'resources', 'media', 'ok.png'))
+
+__confirmTmrAdded__ = True if __addon__.getSetting('confirmTmrAdded').upper() == 'TRUE' else False
+
+OSD = xbmcgui.Dialog()
+OSDProgress = xbmcgui.DialogProgress()
+
+def getSetting(setting):
+    return __addon__.getSetting(setting)
+
+def getDateFormat():
+    df = xbmc.getRegion('dateshort')
+    tf = xbmc.getRegion('time').split(':')
+
+    try:
+        # time format is 12h with am/pm
+        return df + ' ' + tf[0][0:2] + ':' + tf[1] + ' ' + tf[2].split()[1]
+    except IndexError:
+        # time format is 24h with or w/o leading zero
+        return df + ' ' + tf[0][0:2] + ':' + tf[1]
+
+def notifyLog(message, level=xbmc.LOGNOTICE):
+    xbmc.log('[%s] %s' % (__addonid__, message.encode('utf-8')), level)
+
+def notifyOSD(header, message, icon=__IconDefault__, time=5000):
+    OSD.notification(header.encode('utf-8'), message.encode('utf-8'), icon, time)
+
+def date2timeStamp(date, format=getDateFormat()):
+    try:
+        dtime = datetime.datetime.strptime(date, format)
+    except TypeError:
+        try:
+            dtime = datetime.datetime.fromtimestamp(time.mktime(time.strptime(date, format)))
+        except ValueError:
+            notifyLog('Couldn\'t parse date: %s' % (date), xbmc.LOGERROR)
+            notifyOSD(__LS__(30000), __LS__(30020), icon=__IconAlert__)
+            return False
+    except Exception:
+        notifyLog('Couldn\'t parse date: %s' % (date), xbmc.LOGERROR)
+        notifyOSD(__LS__(30000), __LS__(30020), icon=__IconAlert__)
+        return False
+    return int(time.mktime(dtime.timetuple()))
+
+def readTimerStrings():
+    xbmc.sleep(1000)
+    timers = []
+    for prefix in ['t0:', 't1:', 't2:', 't3:', 't4:', 't5:', 't6:', 't7:', 't8:', 't9:']:
+        if xbmc.getInfoLabel('Skin.String(%s)' % (prefix + 'date')) != '':
```

Comment:
I can do that but skinners have no chance to access the data if they are stored outside the skin. Doing this way I have to read them from settings/addon data and store them back into the skin again anyway otherwise they are not passed thru the skin. Yes it's skin related stuff. I've talked about that with the skinners in february this year. There's a [long thread](http://www.kodinerds.net/index.php/Thread/48312-Wir-suchen-einen-Skripter/) about this.